### ISSUE TITLE
Need to use toInteger() to match id ? 

### ISSUE THREAD

**Issue by tomzeppenfeldt** [neo4j/neo4j#11822]
- Neo4j 3.2.6
- Ubuntu
- :7474/browser/ 

## Bug report

I get an unexpected

```
Neo.ClientError.Statement.TypeError
Expected entity id to be an integral value
```

### Steps to reproduce

`:param ids:[185266]`

result : 
```
{
  "ids": [
    185266
  ]
}
```

```
match (n) where id(n) in {ids}
return (n)
```

result

```
Neo.ClientError.Statement.TypeError
Expected entity id to be an integral value
```

### Workaround

```
match (n) where toInteger(id(n)) in {ids}
return (n)
```

returns the node correctly

### Expected behavior
node should be also returned without the toInteger() function

**Comment by SaschaPeukert** [comment]
@ophileon Thanks for reporting. I tried to replicate the problem according to your steps but couldn't get the error. I tried it with 3.3.3 and 3.3.5. Can you confirm this bug in 3.3.5? If so, could you please help me to reproduce it? I tried the following:

Started an empty instance and created some nodes:
```
FOREACH (r IN range(0,185267) | CREATE (:User {idx:r}));
```
Then
```
:param ids:[185266]
```
And
```
match (n) where id(n) in {ids}
return (n)
```

**Comment by tomzeppenfeldt** [comment]
@SaschaPeukert 
checked again.. turns out that is in 3.2.6    
sorry for the confusion.

**Comment by SaschaPeukert** [comment]
No problem, thanks for checking. Now I could reproduce it in 3.2.11 as well. We will look further into it!

**Comment by SimonThordal** [comment]
@ophileon I've noticed that parameters need to be set using `:param ids => [12345]` to be cast as integers whereas using `:` gives you floats. Could this be it?

**Comment by tomzeppenfeldt** [comment]
@SimonThordal  I just tried it in 3.3.3  Community edition and I can reproduce the issue.

3.3.3 does  not accept 
`:param ids => [3] `

in 3.5.0-alpha01   (the one that comes with Neo4j Desktop) 

it works with both

`:param ids => [3] ` and `:param ids : [3] `

which is the correct behaviour I think.

**Comment by tomzeppenfeldt** [comment]
In 3.3.5 it works fine .. no error.

**Comment by SaschaPeukert** [comment]
I just tried it again and It seems to be fixed in 3.2.12
